gradle/javadoc.gradle

```diff
@@ -18,17 +18,18 @@ import static org.gradle.util.CollectionUtils.single
 
 
 def javaMainProjects = subprojects.findAll { !['util'].contains(it.name) }
-task docs(type: Javadoc) {
-    destinationDir = new File(rootDir, 'build/docs')
-    javaMainProjects.each { proj ->
-        proj.tasks.withType(Javadoc).each { javadocTask ->
-            source += javadocTask.source
-            classpath += javadocTask.classpath
-            excludes += javadocTask.excludes
-            includes += javadocTask.includes
-            dependsOn +=javadocTask.dependsOn
-        }
-    }
+
+task docs {
+    dependsOn subprojects.collect { it.tasks.withType(Javadoc) }
+    dependsOn subprojects.collect { it.tasks.withType(ScalaDoc) }
+}
+
+def buildDocsUrl(String version, String packageName) {
+    def docVersion = version.split('-').head()
+    def versionList = docVersion.split('\\.')
+    def patch = (versionList[1] as int) - 1
+    docVersion = patch >= 0 ? "${versionList[0]}.${patch}" : '4.0'
```

Comment:
Should this be hardcoded to `4.0`?

Reply:
Nit: I need to update the patch variable name to minor.

4.0 is the lowest version supported so its hardcoded as the bottom version.  The purpose of this is to take the current version (4.0.0-SNAPSHOT) and reduces the minor version by 1.  If the minor version is lower than 0, then we must be on 4.0.